src/link/error/index.ts

```diff
@@ -44,9 +44,14 @@ export function onError(errorHandler: ErrorHandler): ApolloLink {
       try {
         sub = forward(operation).subscribe({
           next: (result) => {
-            if (result.errors) {
+            const handler = operation.client["queryManager"].incrementalHandler;
+            const errors =
+              handler?.isIncrementalResult(result) ?
```

Comment:
```suggestion
              handler.isIncrementalResult(result) ?
```

This should be non-nullable so no need for the optional chaining 🙂